File: src/589. N-ary Tree Preorder Traversal.py
# 难度-简单
# 简单题，这难度认真的吗。。。
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        ans = []
        def dfs(node):
            if node is None:
                return
            ans.append(node.val)
            for child in node.children:
                dfs(child)
        dfs(root)
        return ans

File: EinstLeet/src/forOffer/589. N-ary Tree Preorder Traversal.cpp
//难度-简单
//简单题，这难度认真的吗。。。
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    void dfs(const Node* n, vector<int>& ans) {
        if(n == nullptr) return;
        ans.push_back(n->val);
        for(auto & c : n->children) dfs(c, ans);
    }
    vector<int> preorder(Node* root) {
        vector<int> ans;
        dfs(root, ans);
        return ans;
    }
};